src/ca11y/index.js

```diff
@@ -69,19 +153,49 @@ class Ca11y {
     }, this.props.transitionDuration)
   }
 
+  /**
+   * Key Down event handler for the associated input.
+   * When the key is 'ESC', the datepicker should close.
+   * @return { Void }
+   **/
   onKeydown(e) {
     if (e.keyCode === keys.esc) this.close()
   }
 
-  setState(state, silent) {
-    this.state = Object.assign({}, this.state, state)
-    if (!silent) this.render()
+  /**
+   * Change event handler for the associated input.
+   * When the value changes and isn't empty, attempt to parse it. If parsing
+   * succeeds, set Ca11y to the resulting date.
+   * @return { Void }
+   **/
+  onInputChange(e) {
+    if (e.target.value) {
+      const result = this.props.parse(e.target.value)
+
+      if (result.valid) {
+        this.setDate(result.date)
+      } else {
+        this.setDate(today) // show current month
+        // set state to NOT SELECTED
```

Comment:
Not quite done with this piece, but Dan and I decided how to handle the case where a user types invalid data into the input. We will be adding a UI state for "no date is currently selected".